src/app/mesh/qgsmeshlayerproperties.cpp

```diff
@@ -229,11 +230,27 @@ void QgsMeshLayerProperties::addDataset()
   settings.setValue( QStringLiteral( "lastMeshDatasetDir" ), openFileInfo.absolutePath(), QgsSettings::App );
   QFile datasetFile( openFileString );
 
+  bool isTemporalBefore = mMeshLayer->dataProvider()->temporalCapabilities()->hasTemporalCapabilities();
   bool ok = mMeshLayer->dataProvider()->addDataset( openFileString );
   if ( ok )
   {
+    if ( !isTemporalBefore && mMeshLayer->dataProvider()->temporalCapabilities()->hasTemporalCapabilities() )
+    {
+      mMeshLayer->temporalProperties()->setDefaultsFromDataProviderTemporalCapabilities(
+        mMeshLayer->dataProvider()->temporalCapabilities() );
+
+      if ( ! mMeshLayer->temporalProperties()->referenceTime().isValid() )
+      {
+        QDateTime referenceTime = QgsProject::instance()->timeSettings()->temporalRange().begin();
+        if ( !referenceTime.isValid() ) // If project reference time is invalid, use current date
+          referenceTime = QDateTime( QDate::currentDate(), QTime( 0, 0, 0, Qt::UTC ) );
+        mMeshLayer->temporalProperties()->setReferenceTime( referenceTime, mMeshLayer->dataProvider()->temporalCapabilities() );
```

Comment:
ins't this code snipped already used somewhere else so it can be a new private function?

Reply:
here : https://github.com/qgis/QGIS/blob/e9dd602da65cdd4c658e4e9b829f8f3fbd59ac40/src/app/qgisapp.cpp#L5487
but the not exactly the same ans I think it is difficult to have an efficient refactoring.

Reply:
ok